aiosmtpd/tests/test_smtp.py

```diff
@@ -202,1404 +211,1420 @@ class SleepingHeloHandler:
-        with SMTP(*self.address) as client:
-            client.ehlo("example.com")
-            code, response = client.docmd("AUTH LOGIN")
-            self.assertEqual(code, 334)
-            self.assertEqual(response, b"VXNlciBOYW1lAA==")
-            code, response = client.docmd('=')
-            self.assertEqual(code, 334)
-            self.assertEqual(response, b"UGFzc3dvcmQA")
-            code, response = client.docmd('=')
-            assert_auth_success(self, code, response)
-            self.assertEqual(auth_peeker.login, None)
-            self.assertEqual(auth_peeker.password, None)
-            code, response = client.mail("alice@example.com")
-            self.assertEqual(self.handler.session.login_data, b"")
-
-    def test_auth_login_abort_login(self):
-        with SMTP(*self.address) as client:
-            client.ehlo("example.com")
-            code, response = client.docmd("AUTH LOGIN")
-            self.assertEqual(code, 334)
-            self.assertEqual(response, b"VXNlciBOYW1lAA==")
-            code, response = client.docmd('*')
-            self.assertEqual(code, 501)
-            self.assertEqual(response, b"Auth aborted")
-
-    def test_auth_login_abort_password(self):
-        auth_peeker.return_val = False
-        with SMTP(*self.address) as client:
-            client.ehlo("example.com")
-            code, response = client.docmd("AUTH LOGIN")
-            self.assertEqual(code, 334)
-            self.assertEqual(response, b"VXNlciBOYW1lAA==")
-            code, response = client.docmd('=')
-            self.assertEqual(code, 334)
-            self.assertEqual(response, b"UGFzc3dvcmQA")
-            code, response = client.docmd('*')
-            self.assertEqual(code, 501)
-            self.assertEqual(response, b"Auth aborted")
-
-    def test_auth_custom_mechanism(self):
-        auth_peeker.return_val = False
-        with SMTP(*self.address) as client:
-            client.ehlo("example.com")
-            code, response = client.docmd("AUTH NULL")
-            assert_auth_success(self, code, response)
+    # This was a bug, and it's already fixed since 3.6 (see bpo below)
+    # Since we now only support >=3.6, there is no point emulating this bug.
+    # Rather, we test that bug is fixed.
+    #
+    # # Test the workaround http://bugs.python.org/issue27931
+    # @patch('email._header_value_parser.AngleAddr.addr_spec',
+    #        side_effect=IndexError, new_callable=PropertyMock)
+    # def test_mail_fail_parse_email(self, addr_spec):
+    #     self.client.helo('example.com')
+    #     self.client.assert_cmd_resp(
+    #         'MAIL FROM: <""@example.com>',
+    #         (501, b'Syntax: MAIL FROM: <address>')
+    #     )
+    def test_27931fix_smtp(self, client):
+        self._helo(client)
+        resp = client.docmd('MAIL FROM: <""@example.com>')
+        assert resp == S.S250_OK
+        resp = client.docmd('RCPT TO: <""@example.org>')
+        assert resp == S.S250_OK
```

Comment:
... rather, we check the fix.

(Which, after some thinking, is totally redundant because it's already tested in that looong list of valid email addresses.)